### PR TITLE
[Doc - Silabs] Update docs to point the SMG docs

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
Update some documentation files in the repo to point silabs's users to additional documentation tied with our stable silabs matter releases offering

### CODE REVIEW

**Review by selissia — APPROVED** [review]

File: docs/guides/README.md [hunk 1]
```diff
@@ -18,8 +18,9 @@
 -   [nRF Connect - Software Update](./nrfconnect_examples_software_update.md)
 -   [NXP - Android Commissioning](./nxp_k32w_android_commissioning.md)
 -   [NXP - Linux Examples](./nxp_imx8m_linux_examples.md)
+-   [Silicon Labs - SMG Documentation](https://github.com/SiliconLabs/matter/tree/v0.3.0/docs/silabs))
```
Comment (selissia) [inline docs/guides/README.md:21]: There's an extra ) at the end

**Review by selissia — APPROVED** [review]

File: docs/guides/README.md (hunk 1 above)
Comment (jmartinez-silabs) [reply docs/guides/README.md]: thanks

File: docs/guides/silabs_efr32_building.md [hunk 2]
```diff
@@ -13,8 +13,18 @@ The full list of examples can be found
 [here](https://github.com/project-chip/connectedhomeip/tree/master/examples).
 The GitHub repository also includes tools for testing your Matter project.
 
+## Silicon Labs Matter Github (SMG)
+
+Silicon Laboratories now maintains a public matter GitHub repo with frequent
+releases thoroughly tested and validated. Developers looking to develop matter
+products with silabs hardware are encouraged to use our latest release,
+currently 0.3.0, with added tools and documentation.
+[Silabs Matter Github](https://github.com/SiliconLabs/matter/releases/tag/v0.3.0)
```
Comment (rerasool) [inline docs/guides/silabs_efr32_building.md:22]: I'd make this the following so we don't have to maintain it:

```
... are encouraged to use our latest release, with added tools and documentation.
[Silabs Matter Github releases](https://github.com/SiliconLabs/matter/releases)
```

File: docs/guides/silabs_efr32_building.md [hunk 3]
```diff
@@ -13,8 +13,18 @@ The full list of examples can be found
...
+
+The complete Silabs documentation can be found here
+[SMG Documentation](https://github.com/SiliconLabs/matter/tree/v0.3.0/docs/silabs)
```
Comment (rerasool) [inline docs/guides/silabs_efr32_building.md:25]: This link always points to our latest readme file (default branch is updated to latest release) so we don't have to maintain it:
`[SMG Documentation](https://github.com/SiliconLabs/matter#readme)`

**Review by andy31415 — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

**Pull request merged**